portfolio/src/main/webapp/jquery_functions.js

```diff
@@ -0,0 +1,31 @@
+/*
+* Cycles between  different background images, triggered by mouse click
+*/
+
+let img_arr = new Array(5);
+img_arr = ["../images/mountbierstadt.jpg", 
+           "../images/mountyale.jpg",
+           "../images/gardenofgods.jpg",
+           "../images/lakedillon.jpg",
+           "../images/mountelbert.jpg"
+           ];
+labels = ["the summit of Mount Bierdstadt.", 
+          "the summit of Mount Yale.", 
+          "the Garden of the Gods.", 
+          "Lake Dillon, seen from the vantage point of driving along Dillon Dam road.",
+          "the summit of Mount Elbert, the tallest point in Colorado."];
+
+
+$(document).ready(function(){
+    // if change background button is clicked, background image for the body is changed. 
+    // Appropriate image label is displayed.
+  let index = 0;
+  $("#change-background").click(function(){
+      index++;
+      if (index == img_arr.length){
+        index = 0;
+      }
+      $("body").css("background-image", "url(" + img_arr[index] + ")");
+      $("#background-caption").html("This image is of "+ labels[index]);
+  })
+});
```

Comment:
Has anything talked about unit tests yet? It seems like this could benefit from a test to ensure it behaves as you expect it to.

Reply:
Unfortunately I don't think we cover unit testing until Week 5 - Test Driven Development. I do have some experience with unit testing (although not for JavaScript) so I could try to make some unit tests now if you think that would be a good idea.

Reply:
Yea, why don't you spend an hour or two trying to get them set up?

If we can I'd love to be testing early--it makes it easier to catch issues.

Reply:
I don't know what framework was used in the development of the bare bones web app they provided, and thus am a bit stuck on installing a testing framework in the correct way. I think for week 5 they have the environment set up for JUnit testing for Java, but nothing set up for JavaScript.

Reply:
I think this is okay if you get stuck or do something different than what they will go over.

Reply:
Sounds good.